src/tests/BuildWasmApps/Wasm.Build.Tests/WasmBuildAppTest.cs

```diff
@@ -85,6 +85,89 @@ public void Bug49588_RegressionTest_NativeRelinking(BuildArgs buildArgs, RunHost
                         extraProperties: "<WasmBuildNative>true</WasmBuildNative>",
                         dotnetWasmFromRuntimePack: false);
 
+        [Theory]
+        [BuildAndRun]
+        public void PropertiesFromRuntimeConfigJson(BuildArgs buildArgs, RunHost host, string id)
+        {
+            buildArgs = buildArgs with { ProjectName = $"runtime_config_{buildArgs.Config}_{buildArgs.AOT}" };
+            buildArgs = ExpandBuildArgs(buildArgs);
+
+            string programText = @"
+                using System;
+                using System.Runtime.CompilerServices;
+
+                var config = AppContext.GetData(""test_runtimeconfig_json"");
+                Console.WriteLine ($""test_runtimeconfig_json: {(string)config}"");
+                return 42;
+            ";
+
+            string runtimeConfigTemplateJson = @"
+            {
+                ""configProperties"": {
+                  ""abc"": ""4"",
+                  ""test_runtimeconfig_json"": ""25""
+                }
+            }";
+
+            BuildProject(buildArgs,
+                        initProject: () =>
+                        {
+                            File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), programText);
+                            File.WriteAllText(Path.Combine(_projectDir!, "runtimeconfig.template.json"), runtimeConfigTemplateJson);
+                        },
+                        id: id,
+                        dotnetWasmFromRuntimePack: !(buildArgs.AOT || buildArgs.Config == "Release"));
+
+            RunAndTestWasmApp(buildArgs, expectedExitCode: 42,
+                                test: output => Assert.Contains("test_runtimeconfig_json: 25", output), host: host, id: id);
+        }
+
+        [Theory]
+        [BuildAndRun]
+        public void PropertiesFromCsproj(BuildArgs buildArgs, RunHost host, string id)
+        {
+            buildArgs = buildArgs with { ProjectName = $"runtime_config_csproj_{buildArgs.Config}_{buildArgs.AOT}" };
+            buildArgs = ExpandBuildArgs(buildArgs);
```

Comment:
Use `ExpandBuildArgs(buildArgs, extraProperties: "<ThreadPoolMaxThreads>20</ThreadPoolMaxThreads>");`

Reply:
updated the test.